helpers/lib/resourceHints.js

```diff
@@ -30,6 +31,11 @@ function addResourceHint(globals, path, state, type, cors) {
         throw new Error('Invalid path provided. path should be a non empty string');
     }
     path = path.trim();
+    try {
+        path = URL.parse(path).format();
+    } catch (error) {
+        throw new Error(`Invalid value (resource URL) provided: [${path}]`)
```

Comment:
You may want to use you may want just to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent

If you really need to validate the url. There is https://github.com/bigcommerce/paper-handlebars/blob/master/helpers/lib/common.js#L4

Reply:
Using the `common` utilities, in this case, is not an option given it does not take into account valid relative URI.

```js
❯ node
Welcome to Node.js v14.20.0.
Type ".help" for more information.
> const common = require('./helpers/lib/common')
undefined
> common.isValidURL('/partial.js?tm=1234567890')
false
>
```

`decodeURIComponent` is not an option either because if `URL.parse` when given an already encoded URL does not double encode it.

```js
> URL.parse(decodeURIComponent('https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=block')).format()
'https://fonts.googleapis.com/css?family=Karla:400%7CMontserrat:400,500,700&display=block'
```